spec/ics-24-host-requirements/README.md

```diff
@@ -0,0 +1,103 @@
+---
+ics: 24
+title: Host State Machine Requirements
+stage: draft
+category: ibc-core
+requires: 2
+required-by: 2, 3, 4, 5, 18
+author: Christopher Goes <cwgoes@tendermint.com>
+created: 2019-04-16
+modified: 2019-04-29
+---
+
+# Synopsis
+
+This specification defines the minimal set of interfaces which must be provided and properties which must be fulfilled by a blockchain & state machine hosting an IBC handler (implementation of the interblockchain communication protocol).
+
+# Specification
+
+## Motivation
+
+IBC is designed to be a common standard which will be hosted by a variety of blockchains & state machines and must clearly define the requirements of the host.
+
+## Definitions
+
+`RootOfTrust` is as defined in [ICS 2](../ics-2-consensus-requirements).
+
+## Desired Properties
+
+IBC should require as simple an interface from the underlying state machine as possible to maximize the ease of correct implementation.
+
+## Technical Specification
+
+### Keys, identifiers, separators
+
+A `Key` is a bytestring used as the key for an object stored in state. Keys MUST contain only alphanumeric characters and the separator `/`.
```

Comment:
Why do the keys & identifiers have to contain only the alphanumeric characters and slashes? In this case, it will be not able to set the arbitrary binary encoded identifiers, for example, lightclient identifier generated from the ROT hash.

Reply:
Either all keys need to be fixed-length, with fixed-length components, or we need a specific separator which cannot be included in identifiers

Reply:
Most of my concern with the variable key is that we are no longer able to use arbitrary bytestring for the identifiers... variable length can be easily simulated with the length prefix method I think.

Reply:
Ok... we need a method to prevent ambiguity between identifiers with a separator in them. That would require specifying multiple lengths at the beginning of a key, one for each separator. Is that what you're suggesting?

Alternatively, we could standardize identifier lengths, which I think is not that bad for connections / channels, but I wonder if it might be bad in other cases...

Why is requiring that identifiers use a restricted subset of ASCII a problem?